core/src/main/kotlin/behavior/channel/TextChannelBehavior.kt

```diff
@@ -69,18 +72,24 @@ interface TextChannelBehavior : PrivateThreadParentChannelBehavior {
     suspend fun startPublicThread(
         name: String,
         archiveDuration: ArchiveDuration = ArchiveDuration.Day,
-        reason: String? = null
-
+        reason: String? = null,
     ): TextChannelThread {
-        return unsafeStartThread(name, archiveDuration, ChannelType.PublicGuildThread, reason) as TextChannelThread
+        return unsafeStartThread(
+            name,
+            archiveDuration,
+            ChannelType.PublicGuildThread
+        ) {
+            this.reason = reason
+        } as TextChannelThread
```

Comment:
remove the reason from here, add the builder as a part of the function
so: 
```kt 
suspend fun startPublicThread(
        name: String,
        archiveDuration: ArchiveDuration = ArchiveDuration.Day,
        builder: StartThread.() -> Unit
    ): TextChannelThread
```
and then pass it to unsafe